src/dials/command_line/combine_experiments.py

```diff
@@ -187,649 +187,215 @@
-                    sys.exit(
-                        "Model didn't match reference within required tolerance for experiment {} in {}:"
-                        "\n{}\nAdjust tolerances or set compare_models=False to ignore differences.".format(
-                            index, path, str(e)
-                        )
-                    )
-
-                # Rewrite imageset_id, if the experiment has and imageset
-                if exp.imageset and "imageset_id" in sub_ref:
-                    # Get the index of the imageset for this experiment and record how it changed
-                    new_imageset_id = experiments.imagesets().index(
-                        experiments[-1].imageset
-                    )
-                    old_imageset_id = exps.imagesets().index(exp.imageset)
-                    imageset_result_map[old_imageset_id] = new_imageset_id
-
-                    # Check for invalid(?) imageset_id indices... and leave if they are wrong
-                    if len(set(sub_ref["imageset_id"])) != 1:
-                        logger.warning(
-                            "Warning: Experiment %d reflections appear to have come from multiple imagesets - output may be incorrect",
-                            i,
-                        )
-                    else:
-                        sub_ref["imageset_id"] = flex.int(len(sub_ref), new_imageset_id)
-
-                reflections.extend(sub_ref)
-
-                global_id += 1
-
-            # Include unindexed reflections, if we can safely remap their imagesets
-            if "imageset_id" in reflections:
-                unindexed_refs = refs.select(refs["id"] == -1)
-                for old_id in set(unindexed_refs["imageset_id"]):
-                    subs = unindexed_refs.select(
-                        unindexed_refs["imageset_id"] == old_id
-                    )
-                    subs["imageset_id"] = flex.int(
-                        len(subs), imageset_result_map[old_id]
-                    )
-                    reflections.extend(subs)
-
-        if (
-            params.output.min_reflections_per_experiment is not None
-            and skipped_expts_min_refl > 0
-        ):
-            print(
-                "Removed {} experiments with fewer than {} reflections".format(
-                    skipped_expts_min_refl, params.output.min_reflections_per_experiment
-                )
+            for cluster in clusters
+        ]
+        if reflections:
+            output_reflections_list = [
+                reflections.select_on_experiment_identifiers(cluster.lattice_ids)
+                for cluster in clusters
+            ]
+            for table in output_reflections_list:
+                table.reset_ids()
+        for i_cluster, _ in enumerate(output_experiments_list):
+            exp_name = os.path.splitext(experiments_filename)[0] + (
+                "_cluster%d.expt" % (n_clusters - i_cluster)
```

Comment:
We had a discussion in #2311 about this line, line 219 and line 245. As of now, the clusters are numbered from smallest to largest, with the first few usually missing because they are rejected in lines 223–224. This appears to be a bug or, at least, an unexpected behavior. I started #2318 to tackle this issue (and to increase some readability), but this will be heavily superseded by this PR, so I'll pull out. Nevertheless, I still wonder whether this cluster indexing procedure could be changed to index clusters according to decreasing size and starting from 0.

Reply:
Hi @Baharis , my intention here was not to override your work in #2318 in any way. I think it's work keeping the cluster order renaming as a separate piece of work for clarity, then I can just rebase these reorganisationalchanges on any work you do there.

Reply:
Alright, in this case I will re-test my PR and see if it can be quickly merged. However, if there are any problems or delays with my PR, feel free to merge first and I'll return to the issue of ordering whenever I'll find some time.